src/log.rs

```diff
@@ -139,10 +139,28 @@ impl EspLogger {
 
     #[cfg(not(all(esp_idf_version_major = "4", esp_idf_version_minor = "3")))]
     fn should_log(record: &Record) -> bool {
+        use crate::private::mutex::Mutex;
+        use alloc::collections::BTreeMap;
+        use once_cell::sync::Lazy;
+
+        // esp-idf `function esp_log_level_get` builds a cache using the address
+        // of the target and not doing a string compare.  This means we need to
+        // build a cache of our own mapping the string value to a consistant
+        // c-string value.
+        static TARGET_CACHE: Lazy<Mutex<BTreeMap<String, CString>>> = Lazy::new(|| Mutex::new(BTreeMap::new()));
         let level = Newtype::<esp_log_level_t>::from(record.level()).0;
-        let max_level = unsafe {
-            esp_log_level_get(b"rust-logging\0" as *const u8 as *const _) // TODO: use record target?
+        let mut cache = TARGET_CACHE.lock();
+        let ctarget = match cache.get(record.target()) {
+            Some(value) => value,
+            None => {
+                // println!("EspLogger::should_log: add to cache target={}", record.target());
+                let value = CString::new(record.target()).unwrap();
```

Comment:
I think there was a way to do the insert + get with a single map op.

Reply:
found it